src/main/java/com/parpiiev/time/controllers/ActivityController.java

```diff
@@ -0,0 +1,205 @@
+package com.parpiiev.time.controllers;
+
+import com.parpiiev.time.exceptions.activity.ActivityAlreadyExistsException;
+import com.parpiiev.time.services.interfaces.ActivityService;
+import com.parpiiev.time.services.interfaces.CategoryService;
+import com.parpiiev.time.utils.dto.ActivityDTO;
+import com.parpiiev.time.utils.dto.CategoryDTO;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static com.parpiiev.time.controllers.Paths.*;
+
+@RequiredArgsConstructor
+@Controller
+public class ActivityController {
+
+    private static final Logger log = LoggerFactory.getLogger(ActivityController.class);
+
+    private final ActivityService<ActivityDTO> activityService;
+    private final CategoryService<CategoryDTO> categoryService;
+
+//    @GetMapping(ALL_ACTIVITIES_PAGE)
+//    public String showPage(Model model) {
+//        log.debug("Start ActivityController GET showPage");
+//
+//        List<CategoryDTO> category = new ArrayList<>();
+//        model.addAttribute("category", category);
+//        List<CategoryDTO> categories =  categoryService.getAll();
+//        model.addAttribute("categories", categories);
+//
+//        List<ActivityDTO> activities = activityService.getAll();
+//        model.addAttribute("activities", activities);
+//        model.addAttribute("activityDTO", new ActivityDTO());
+//
+//        return ALL_ACTIVITIES_FILE;
+//    }
+
+    @GetMapping(ALL_ACTIVITIES_PAGE)
+    public String listActivities(Model model, HttpSession session,
+                                 @RequestParam("page") Optional<Integer> page,
+                                 @RequestParam("size") Optional<Integer> size,
+                                 @RequestParam(defaultValue = "id") Optional<String> sortBy,
```

Comment:
if you have `defaultValue` then there is no point in keeping `sortBy` as Optional
```suggestion
                                 @RequestParam(defaultValue = "id") String sortBy,
```